tools-local/unix/tool-helper.sh

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env bash
+
+# Provides helper functions.
+
+# Gets name of the operating system.
+# Exit 1 if unable to get operating system name.
+get_os_name()
+{
+    osName="$(uname -s)"
+
+    if [ $? -ne 0 ] || [ -z "$osName" ]; then
+        echo "Unable to determine the name of the operating system."
+        exit 1
+    fi
+
+    if echo "$osName" | grep -iqF "Darwin"; then
+        osName="OSX"
+    else
+        osName="Linux"
+    fi
+
+    echo "$osName"
+}
+
+# Gets the path to the folder corresponding to the specified tool name in Tools/downloads folder under repository root.
+# Exit 1 if unable to determine the path.
+get_repository_tools_downloads_folder()
+{
+    if [ -z "$1" ]; then
+        echo "Argument passed as repository-root is empty. Please provide a non-empty string."
+        usage
+        exit 1
+    fi
+
+    if [ -z "$2" ]; then
```

Comment:
Would it make sense to encapsulate this in a helper method?  I feel like this file has as much arg checking logic as other logic.

validate_required_arg $2 "tool name"

another option to shorten more would be something like
toolName = get_required_arg $2 "tool name"